ouroboros-consensus/test-util/Test/Util/FS/Sim/MockFS.hs

```diff
@@ -542,6 +580,56 @@ hGetSome err@ErrorHandling{..} h n =
       , fsLimitation  = True
       }
 
+-- | Thread safe version of 'hGetSome', which doesnt modify or read the file
+-- offset.
+hGetSomeAt :: CanSimFS m
+           => ErrorHandling FsError m
+           -> Handle'
+           -> Word64
+           -> Int64
+           -> m ByteString
+hGetSomeAt err@ErrorHandling{..} h n o =
+    withOpenHandleStateRead err h $ \fs OpenHandle{..} -> do
+      file <- checkFsTree err $ FS.getFile openFilePath (mockFiles fs)
+      let fsize = fromIntegral (BS.length file) :: Word64
+      case (openPtr, sign64 o)  of
+        (RW r _ _, Positive o') -> do
+          unless r $ throwError (errNoReadAccess openFilePath "write")
+          let bs = BS.take (fromIntegral n) . BS.drop (fromIntegral o) $ file
+          -- This is the same fsLimitation we get when we seek past the end of
+          -- EOF, in AbsoluteSeek mode.
+          when (o' > fsize) $ throwError (errPastEnd openFilePath)
```

Comment:
Why do we need to check whether the current offset *of the handle* is past the end of the file?

Reply:
This is not the offset of the file, but the offset we're reading. It is an error to give an offset bigger than the file size, so I have an fslimitation here.

Reply:
But `o` is the offset we're reading. `o'` is the current offset of the file, which should not matter for `pread`, I would assume.

Reply:
`o'` is just the absolute value of `o`. The file offset would be given by `RW r _ o`, but we actually ignore the third argument of `RW`. So we don't have scope to the file offset anywhere in this function.

Reply:
:man_facepalming: Sorry, I misread. Probably because in my mind the offset was already a `Word64` (as I suggested before).